File: clients/c/examples/debugger.c
/* Example: breakpoints and stepping */
#include "bashclient.h"
#include <stdio.h>

int main(int argc, char **argv)
{
    const char *path = argc > 1 ? argv[1] : "/tmp/bash-server/sock";
    const char *token = argc > 2 ? argv[2] : "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";

    bc_client_t *c = bc_connect(path);
    if (!c || bc_auth(c, token) != BC_OK) { bc_close(c); return 1; }

    bc_debug_enable(c);
    int bp_id = bc_debug_add_breakpoint(c, "command", "echo", -1, NULL);
    printf("Breakpoint set: id=%d\n", bp_id);

    bc_debug_status_t status;
    bc_debug_status(c, &status);
    printf("Debug active=%d mode=%s bps=%d\n",
           status.active, status.mode ? status.mode : "", status.breakpoints);
    bc_debug_status_free(&status);

    bc_debug_remove_breakpoint(c, bp_id);
    bc_debug_disable(c);
    bc_close(c);
    return 0;
}

File: bash-server/clients/java/examples/Debugger.java
package org.gnu.bash.client.examples;

import org.gnu.bash.client.BashClient;

public class Debugger {
    public static void main(String[] args) throws Exception {
        String path = args.length > 0 ? args[0] : "/tmp/bash-server/sock";
        String token = args.length > 1 ? args[1] : "a".repeat(64);

        try (BashClient client = BashClient.connect(path)) {
            client.auth(token);
            client.debug.enable();
            int bpId = client.debug.addBreakpoint("command", "echo", -1, null);
            System.out.println("Breakpoint set: id=" + bpId);
            client.debug.on("break_hit", ev -> System.out.println("[BREAK] " + ev));
            client.eval("echo breakpoint_test");
            client.debug.removeBreakpoint(bpId);
            client.debug.disable();
        }
    }
}
